mergebom.py

```diff
@@ -15,224 +15,99 @@
-        "-m",
-        "--replace-merged",
-        dest="replace_merged",
-        action="store_false",
-        default=True,
-        help="Remove all boms files after merge (only when using version file)")
-    parser.add_option("-r", "--bom-revision", dest="bom_rev",
-                      default=None, help="Hardware BOM revision")
-    parser.add_option("-w", "--bom-pcb-revision", dest="bom_pcb_ver",
-                      default=None, help="PCB Revision")
-    parser.add_option("-n", "--bom-prj-name", dest="bom_prj_name",
-                      default=None, help="Project names.")
-    parser.add_option(
-        "-t",
-        "--bom-date",
-        dest="bom_prj_date",
-        default=datetime.datetime.today().strftime("%d/%m/%Y"),
-        help="Project date.")
-
-    (options, args) = parser.parse_args()
-
-    # Get logger
+        
+    parser.add_argument('--prj_date', '-date', dest='prj_date', 
+                        help='prj_date', default=None)
+    parser.add_argument('--prj_hw_ver', '-hw_ver', dest='prj_hw_ver', 
+                        help='prj_hw_ver', default=None)
+    parser.add_argument('--prj_license', '-license', dest='prj_license', 
+                        help='prj_license', default=None)
+    parser.add_argument('--prj_name', '-name', dest='prj_name', 
+                        help='prj_name', default=None)
+    parser.add_argument('--prj_name_long', '-name_long', dest='prj_name_long', 
+                        help='prj_name_long', default=None)
+    parser.add_argument('--prj_pcb', '-pcb', dest='prj_pcb', 
+                        help='prj_pcb', default=None)
+    parser.add_argument('--prj_pn', '-pn', dest='prj_pn', 
+                        help='prj_pn', default=None)
+    parser.add_argument('--prj_status', '-status', dest='prj_status', 
+                        help='prj_status', default=None)                    
+    options = parser.parse_args()
+
+    f_list= []
+    if not options.ws == None:
+        file_BOM = cfg.cfg_altiumWorkspace(options.ws, options.csv_file) 	 
+        if len(file_BOM)>0:
+            appo = []
+            for i,v in enumerate(file_BOM):
+                parametri_dict={}
+                appo = file_BOM[i][0]
+                parametri_dict=file_BOM[i][1]
+                options.prj_date=parametri_dict.get('prj_date', None)
+                options.prj_hw_ver=parametri_dict.get('prj_hw_ver',None)
+                options.prj_license=parametri_dict.get('prj_license', None)
+                options.prj_name=parametri_dict.get('prj_name', None)
+                options.prj_name_long=parametri_dict.get('prj_name_long', None)
+                options.prj_pcb=parametri_dict.get('prj_pcb', None)
+                options.prj_pn=parametri_dict.get('prj_pn', None)
+                options.prj_status=parametri_dict.get('prj_status', None)
+                for j,v in enumerate(appo):
+                    f_list.append(appo[j])
+        else:
+            sys.exit
```

Comment:
Qui è lo stesso ragionamento che ti ho fatto in questi giorni sugli if, in questo caso direi che è molto più chiaro verificare prima la condizione di errore, e nel caso si esce, altrimenti eseguo il codice.
Altra nota, che quando si esche con la `sys.exit`, gli devi anche passare il codice di errore, che poi nella shell questo può essere gestito. Infatti non è detto che se chiami la `sys.exit` è un errore, infatti per convezione se esci con `0` non è errore ma va bene, mentre un codice diverso da 0 invece è un errore.
Quindi, nel nostro caso:
`sys.exit(1)` perchè si è verificato un errore.